### PR TITLE
arm: Add minimal support for Arm Cortex-M85

### PR DESCRIPTION
**Pull request by thaoluonguw**
Add minimal support for Arm Cortex-M85:

This cherry picked from https://github.com/gcc-mirror/gcc

ccfd1e7f0d9686aa931e65a04845a7436f85d71c : arm: Add support for Arm Cortex-M85 CPU.

This minimal support excluded pacbti feature which is causing build error in Zephyr SDK.

### CODE REVIEW

**Comment by keith-packard** [comment]
Fixes for pacbti are upstream in gcc and should be easy to backport here; is that something that should be included here?

**Comment by stephanosio** [comment]
> Fixes for pacbti are upstream in gcc and should be easy to backport here; is that something that should be included here?

@keith-packard Given that PACBTI support was added in GCC 13 and the current Zephyr fork is based on GCC 12, wouldn't it be too risky to attempt this (whole feature) backport?

**Comment by keith-packard** [comment]
> > Fixes for pacbti are upstream in gcc and should be easy to backport here; is that something that should be included here?
> 
> @keith-packard Given that PACBTI support was added in GCC 13 and the current Zephyr fork is based on GCC 12, wouldn't it be too risky to attempt this (whole feature) backport?

Sorry, I'd forgotten Zephyr was still on GCC 12.

**Comment by thaoluonguw** [comment]
> Fixes for pacbti are upstream in gcc and should be easy to backport here; is that something that should be included here?

@keith-packard , @stephanosio : we would like to propose minimal support for CM85 at this time. To support pacbti, it requires more backport. I saw that there is planning to upgrade gcc to v14.2 as information in https://github.com/zephyrproject-rtos/sdk-ng/issues/740, until v14.2 is supported, is it Ok to support for CM85 with this minimal support?

File: gcc/doc/invoke.texi [hunk 1]
```diff
@@ -21244,14 +21244,25 @@ The following extension options are common to the listed CPUs:
 
 @table @samp
 @item +nodsp
-Disable the DSP instructions on @samp{cortex-m33}, @samp{cortex-m35p}.
+Disable the DSP instructions on @samp{cortex-m33}, @samp{cortex-m35p} and @samp{cortex-m85}. Also disable the M-Profile Vector Extension (MVE) integer and single precision floating-point instructions on @samp{cortex-m85}.
+
+@item +nopacbti
+Disable the Pointer Authentication and Branch Target Identification Extension
+on @samp{cortex-m85}.
```
Comment (stephanosio) [inline gcc/doc/invoke.texi]: `nopacbti`, which is not a valid option in this version of GCC, should not be mentioned in the man pages.

File: gcc/testsuite/gcc.target/arm/multilib.exp [hunk 2]
```diff
@@ -521,6 +521,11 @@ if {[multilib_config "rmprofile"] } {
 	{-mcpu=cortex-m23 -mfpu=fpv5-d16 -mfloat-abi=soft} "thumb/v8-m.base/nofp"
 	{-mcpu=cortex-m33 -mfpu=fpv5-d16 -mfloat-abi=soft} "thumb/v8-m.main/nofp"
 	{-mcpu=cortex-m7+nofp.dp -mfpu=fpv5-d16 -mfloat-abi=soft} "thumb/v7e-m/nofp"
+	{-mcpu=cortex-m85+nopacbti -mfpu=auto -mfloat-abi=soft} "thumb/v8-m.main/nofp"
+	{-mcpu=cortex-m85+nopacbti+nofp -mfpu=auto -mfloat-abi=soft} "thumb/v8-m.main/nofp"
+	{-mcpu=cortex-m85+nopacbti+nomve -mfpu=auto -mfloat-abi=soft} "thumb/v8-m.main/nofp"
+	{-mcpu=cortex-m85+nopacbti+nodsp -mfpu=auto -mfloat-abi=soft} "thumb/v8-m.main/nofp"
+	{-mcpu=cortex-m85+nopacbti+nomve.fp -mfpu=auto -mfloat-abi=soft} "thumb/v8-m.main/nofp"
```
Comment (stephanosio) [inline gcc/testsuite/gcc.target/arm/multilib.exp:528]: Obviously, these are not going to pass the test; but, we might as well leave it to keep the diff minimal from the original commit.

**Comment by ra-fsp-systems** [comment]
> > Fixes for pacbti are upstream in gcc and should be easy to backport here; is that something that should be included here?
> 
> @keith-packard Given that PACBTI support was added in GCC 13 and the current Zephyr fork is based on GCC 12, wouldn't it be too risky to attempt this (whole feature) backport?

@stephanosio Is there a plan for moving ahead to GCC 13?

**Comment by stephanosio** [comment]
> @stephanosio Is there a plan for moving ahead to GCC 13?

We will likely skip 13 and target 14 (see https://github.com/zephyrproject-rtos/sdk-ng/issues/740#issuecomment-1953895728).

File: gcc/doc/invoke.texi (hunk 1 above)
Comment (thaoluonguw) [reply gcc/doc/invoke.texi]: @stephanosio , thank you for your comment. I agree with you. I will update for commit "arm: Remove pacbti in cortex-m85" to remove pacbti information in man pages.

Comment (thaoluonguw) [reply gcc/doc/invoke.texi]: @stephanosio , I updated in https://github.com/zephyrproject-rtos/gcc/pull/28/commits/d1b1cc57623f51cd53ca01bebdb2cc075aaac505, please help me review for it. Thank you.

**Review by stephanosio — APPROVED** [review]

**Pull request merged**